plugins/woocommerce/src/Internal/Admin/Events.php

```diff
@@ -250,4 +253,21 @@ protected function is_merchant_email_notifications_enabled() {
 		// All checks have passed.
 		return true;
 	}
+
+	/**
+	 *   Refresh transient for the following DataSourcePollers on wc_admin_daily cron job.
+	 *   - PaymentGatewaySuggestionsDataSourcePoller
+	 *   - RemoteFreeExtensionsDataSourcePoller
+	 */
+	protected function possibly_refresh_data_source_pollers() {
+		$completed_tasks = get_option( 'woocommerce_task_list_tracked_completed_tasks' );
+
+		if ( ! in_array( 'payments', $completed_tasks, true ) && ! in_array( 'woocommerce-payments', $completed_tasks, true ) ) {
+			PaymentGatewaySuggestionsDataSourcePoller::get_instance()->read_specs_from_data_sources();
+		}
+
+		if ( ! in_array( 'store_details', $completed_tasks, true ) ) {
```

Comment:
I'm not sure why we need it for `store_details` but we also use Remote Free Extensions data in the `marketing` task. Should we add it here?

Reply:
It's used in `Business Details` step.

![Screen Shot 2023-03-15 at 5 42 42 PM](https://user-images.githubusercontent.com/4723145/225479969-0030b917-26a6-4b1b-9b9e-9f24a30ebf56.jpg)


> but we also use Remote Free Extensions data in the marketing task. Should we add it here?

Good idea! Added in https://github.com/woocommerce/woocommerce/pull/37027/commits/b410aadbf5bd920bf9af20ed7448c0b822946458